api/agent/protocol/json.go

```diff
@@ -3,16 +3,27 @@ package protocol
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io"
 	"net/http"
+
+	"github.com/fnproject/fn/api/models"
 )
 
 // This is sent into the function
 // All HTTP request headers should be set in env
 type jsonio struct {
-	Headers    http.Header `json:"headers,omitempty"`
-	Body       string      `json:"body"`
-	StatusCode int         `json:"status_code,omitempty"`
+	Headers map[string]string `json:"headers,omitempty"`
+	Body    string            `json:"body"`
+}
```

Comment:
why status code no longer required?

Reply:
it's still there, it uses a separate struct now to write output that embeds this one.